kymatio/utils.py

```diff
@@ -0,0 +1,48 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import torch
+
+
+class ScatteringTransformer(BaseEstimator, TransformerMixin):
+
+    def __init__(self, S, signal_shape):
+        """Creates an object that is compatible with the scikit-learn API
+        and implements the .transform method.
+        
+        Parameters
+        ==========
+
+        S: an instance of Scattering1D, Scattering2D or Scattering3D
+            This instance is called by the transformer
+
+        signal_shape: tuple
+            The shape of one sample. Scikit-learn convention is to work with
+            2D matrices only, of shape (n_samples, n_features). Data is
+            delivered in this way and has to be reshaped before transform
+
+        Output
+        ======
+        Y: ndarray, shape (n_samples, n_scattering_features)
+
+        """
+
+        self.scattering = S
+        self.signal_shape = signal_shape
+
+    def fit(self, X=None, y=None):
+        # no fitting necessary
+        return self
+
+    def predict(self, X):
+        
+        X_reshaped = X.reshape((-1,) + self.signal_shape)
+        transformed = self.scattering(
+            torch.from_numpy(X_reshaped).to(self.scattering.device))
```

Comment:
I'm adding the backend numpy asap to avoid using torch with sklearn

Reply:
alright, we'll have to modify this here whenever that is done

Reply:
that's done